test/unit/components/chart/stats.test.js

```diff
@@ -106,5 +107,106 @@ describe('Stats', () => {
         expect(wrapper.find(`#Stat--${statId}`)).to.have.length(1);
       });
     });
+
+    describe('collapse state', () => {
+      afterEach(() => {
+        Stats.__ResetDependency__('useLocalStorage');
+        baseProps.trackMetric.resetHistory();
+      });
+
+      it('should render the stats with correct `isOpened` and `title` props from localStorage', () => {
+        Stats.__Rewire__('useLocalStorage', sinon.stub().returns([
+          {
+            basics: {
+              // collapsed state of stats
+              averageDailyDose: false,
+              carbs: true,
+            },
+          },
+          sinon.stub()
+        ]));
+
+        wrapper.setProps({
+          ...wrapper.props(),
+          stats: [
+            {
+              id: 'carbs',
+              collapsible: false,
+              title: 'Carbs',
+              collapsedTitle: 'Carbs Collapsed',
+            },
+            {
+              id: 'averageDailyDose',
+              collapsible: true,
+              title: 'Daily Dose',
+              collapsedTitle: 'Daily Dose Collapsed',
+            },
+          ],
+        });
+
+        // non-collapsible stats should always be open and use the standard title prop,
+        // even if collapsed state in local storage is `true`
+        const carbStat = wrapper.find('#Stat--carbs').childAt(0);
+        expect(carbStat.props().isOpened).to.be.true;
+        expect(carbStat.props().title).to.equal('Carbs');
+
+        // collapsible stat should follow the collapsed state
+        const dailyDoseStat = () => wrapper.find('#Stat--averageDailyDose').childAt(0);
+        expect(dailyDoseStat().props().isOpened).to.be.true;
+        expect(dailyDoseStat().props().title).to.equal('Daily Dose');
+
+        // Update collapsed state to true
+        Stats.__Rewire__('useLocalStorage', sinon.stub().returns([
+          {
+            basics: {
+              // collapsed state of stats
+              averageDailyDose: true,
+              carbs: true,
+            },
+          },
+          sinon.stub()
+        ]));
+
+        wrapper.setProps();
+
```

Comment:
It seems per the comment above that we should verify that `carbStat` is still opened and title hasn't changed even if the localStorage state reflects a collapsed state.

Reply:
I've added a couple more test assertions to cover this.  Thanks 👍

Reply:
I notice that dailyDoseStat is the return from a function containing a find. Will the `carbStat` reference be properly updated in the new assertions, or would it also need an arrow function to ensure that we're checking the updated value?

Reply:
Let me check.

Reply:
Yes, good catch.  I've updated that one to a function as well.